src/main/java/org/vanilladb/core/util/Timer.java

```diff
@@ -76,7 +91,23 @@ public void reset() {
 		subTimers.clear();
 		componenents.clear();
 	}
+	
+	public void setStartComponentTimer(Object component,long start) {
+		SubTimer timer = subTimers.get(component);
+		if (timer == null) {
+			timer = new SubTimer();
+			subTimers.put(component, timer);
+			componenents.add(component);
+		}
+		timer.setStartTime(start);
+	}
 
+	public void setStopComponentTimer(Object component, long stop) {

```

Comment:
Please change the signature to `public void stopComponentTimer(Object component, long stopTime)` and put this method right after `public void stopComponentTimer(Object component)`.